src/main/drivers/accgyro/accgyro_spi_icm40609.c

```diff
@@ -0,0 +1,766 @@
+}
+
+// Set endianness for sensor data (bit 4 of INTF_CONFIG0, reg 0x4C)
+// true  = Big Endian
+// false = Little Endian
+static void icm40609SetEndianess(const extDevice_t *dev, bool bigEndian)
+{
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+
+    uint8_t reg4C = spiReadRegMsk(dev, ICM40609_REG_INTF_CONFIG0);
+    reg4C &= ~(1 << 4);
+    reg4C |= bigEndian << 4;
+
+    spiWriteReg(dev, ICM40609_REG_INTF_CONFIG0, reg4C);
+}
+
+void icm40609AccInit(accDev_t *acc)
+{
+    acc->acc_1G = 2048; // 16g scale
+    acc->gyro->accSampleRateHz = 1000;
+    
+}
+
+void icm40609GyroInit(gyroDev_t *gyro)
+{
+    const extDevice_t *dev = &gyro->dev;
+    spiSetClkDivisor(dev, spiCalculateDivider(ICM40609_MAX_SPI_CLK_HZ));
+
+    mpuGyroInit(gyro);
+    gyro->accDataReg = ICM40609_ACCEL_DATA_X1_UI;
+    gyro->gyroDataReg = ICM40609_GYRO_DATA_X1_UI;
+
+    setGyroAccPowerMode(dev, false);
+
+    icm40609SetEndianess(dev, true);
+
+    icm40609SelectUserBank(dev, ICM40609_USER_BANK_0);
+    spiWriteReg(dev, ICM40609_REG_GYRO_CONFIG0, ICM40609_GYRO_FS_SEL_2000DPS | ICM40609_GYRO_ODR_8KHZ);
+    gyro->scale = GYRO_SCALE_2000DPS;
+    gyro->gyroRateKHz = GYRO_RATE_8_kHz;
+    gyro->gyroSampleRateHz = 8000;
+
+    spiWriteReg(dev, ICM40609_REG_ACCEL_CONFIG0, ICM40609_ACCEL_FS_SEL_16G | ICM40609_ACCEL_ODR_1KHZ );
+
+    icm40609SetTempFiltBw(dev, ICM40609_TEMP_FILT_BW_4000HZ); // 4000Hz, 0.125ms latency (default)
+    icm40609SetGyroUiFiltOrder(dev, ICM40609_UI_FILT_ORDER_3RD);
+    icm40609SetAccelUiFiltOrder(dev, ICM40609_UI_FILT_ORDER_3RD);
+    icm40609SetGyroDec2M2(dev, true);
+
+    // Set filter bandwidth: Low Latency
+    spiWriteReg(&gyro->dev, ICM40609_REG_GYRO_ACCEL_CONFIG0, 
+                    ICM40609_ACCEL_UI_FILT_BW_LP_TRIVIAL_200HZ_8XODR | 
+                    ICM40609_GYRO_UI_FILT_BW_LP_TRIVIAL_200HZ_8XODR);
+
+    uint16_t gyroHWLpf; // Anti-Alias Filter (AAF) in Hz
+    switch (gyroConfig()->gyro_hardware_lpf) {
+    case GYRO_HARDWARE_LPF_NORMAL:
+        gyroHWLpf = 213;
+        break;
+    case GYRO_HARDWARE_LPF_OPTION_1:
+        gyroHWLpf = 488;
+        break;
+    case GYRO_HARDWARE_LPF_OPTION_2:
+        gyroHWLpf = 997;
+        break;
+#ifdef USE_GYRO_DLPF_EXPERIMENTAL
+    case GYRO_HARDWARE_LPF_EXPERIMENTAL:
+        gyroHWLpf = 1962;
+        break;
+#endif
+        default:
+        gyroHWLpf = 213;
+    }
+
+    icm40609SetGyroAafByHz(dev, true, gyroHWLpf);
+    icm40609SetAccelAafByHz(dev, true, gyroHWLpf);
+
+    icm40609SetGyroNotch(dev, true, ICM40609_GYRO_NF_BW_1449HZ, 1.5f);
+
+    icm40609SetGyroHPF(dev, true, ICM40609_HPF_BW_1, ICM40609_HPF_ORDER_1ST);
+                  
+    // Enable interrupt
+    spiWriteReg(dev, ICM40609_REG_INT_SOURCE0, ICM40609_UI_DRDY_INT1_EN);
+
+    // Set INT1 to pulse mode, push-pull, active high
+    spiWriteReg(dev, ICM40609_REG_INT_CONFIG, 
+        ICM40609_INT1_MODE_PULSED | 
+        ICM40609_INT1_DRIVE_PUSH_PULL | 
+        ICM40609_INT1_POLARITY_ACTIVE_LOW); 
```

Comment:
Sorry, that was my mistake. BF expects active-high ISR ( `EXTIConfig(mpuIntIO, &gyro->exti, NVIC_PRIO_MPU_INT_EXTI, IOCFG_IN_FLOATING, BETAFLIGHT_EXTI_TRIGGER_RISING);` )

Reply:
reverted to ACTIVE HIGH